### PR TITLE
add: `hakuneko-deb`

### PR DESCRIPTION
**Pull request by Diegiwg**
- [x] Edit packagelist
- [x] Add initial pacscript
- [x] [Contact devs](https://github.com/manga-download/hakuneko/issues/5881)
- [x] Add maintainer to pacscript

### CODE REVIEW

**Comment by Diegiwg** [comment]
The package offers versions for multiple architectures. So how can we provide HASH for such architectures?

**Comment by Henryws** [comment]
On your contact devs link, `$name` should be replaced with the name of the actual program and the link to this PR should be replaced with the actual link:
```md
[Pacstall](https://github.com/pacstall/pacstall) is a community-driven AUR-like package manager for Ubuntu. We have an ongoing pull request to add [hakuneko](https://github.com/pacstall/pacstall-programs/pull/3657) to our repository.

We have a few questions for you:
* Would you like to maintain the pacscript (similar to a PKGBUILD) yourself? We can maintain it for you if you decide not to.
* Could you include a section in your documentation showing Pacstall as a valid method of installation for Hakuneko for Debian/Ubuntu users? We could make a pull request for that if you want.
```

**Comment by Henryws** [comment]
> The package offers versions for multiple architectures. So how can we provide HASH for such architectures?

Use `$CARCH` in a case statement like [mold-bin](https://github.com/pacstall/pacstall-programs/blob/77ec6025494a8372cf6013fc6eafe7b79152779d/packages/mold-bin/mold-bin.pacscript#L13)

**Comment by Diegiwg** [comment]
Thanks for the notice about the Issue on the authors. And thanks for the tip for dealing with multiple architectures.

**Comment by Diegiwg** [comment]
@Henryws, could you explain to me the problem with `pre-commit`? I'm not understanding the error.

**Comment by Henryws** [comment]
It basically is just adding a newline that your editor didn't add.

File: packages/hakuneko-deb/hakuneko-deb.pacscript [hunk 1]
```diff
@@ -0,0 +1,32 @@
+name="hakuneko-deb"
+version="8.3.4"
+nightly="20200705.1"
+
```
Comment (Henryws) [inline packages/hakuneko-deb/hakuneko-deb.pacscript:4]:
```suggestion
```

File: packages/hakuneko-deb/hakuneko-deb.pacscript [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    url="https://github.com/manga-download/hakuneko/releases/download/nightly-${nightly}/hakuneko-desktop_${version}_linux_$CARCH.deb"
+    hash="b263aaf4b7bb7309b0dfca6fe0fe8f39ecce303927a52aad097c96aef57fc973"
+    ;;
+  i386)
+    url="https://github.com/manga-download/hakuneko/releases/download/nightly-${nightly}/hakuneko-desktop_${version}_linux_$CARCH.deb"
+    hash="3608ff4e8a3257fffedfa2dcea6501cb6d0d203235a2e4c6e6bbf01b93dce240"
+    ;;
+  *) return 1 ;;
+esac
+
```
Comment (Henryws) [inline packages/hakuneko-deb/hakuneko-deb.pacscript:29]:
```suggestion
```

File: packages/hakuneko-deb/hakuneko-deb.pacscript [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+homepage="https://github.com/manga-download/hakuneko"
+description="HakuNeko is a cross-platform downloader for manga and anime from various websites."
```
Comment (Henryws) [inline packages/hakuneko-deb/hakuneko-deb.pacscript:31]:
```suggestion
description="HakuNeko is a cross-platform downloader for manga and anime from various websites"
```

**Review by Henryws — CHANGES_REQUESTED** [review]

**Review by Henryws — CHANGES_REQUESTED** [review]

File: packages/hakuneko-deb/hakuneko-deb.pacscript [hunk 4]
```diff
@@ -0,0 +1,32 @@
+name="hakuneko-deb"
+version="8.3.4"
+nightly="20200705.1"
+arch=("amd64" "arm64" "armhf" "armv7l" "i386")
+
+case "${CARCH}" in
+  amd64)
+    url="https://github.com/manga-download/hakuneko/releases/download/nightly-${nightly}/hakuneko-desktop_${version}_linux_${CARCH}.deb"
```
Comment (Henryws) [inline packages/hakuneko-deb/hakuneko-deb.pacscript:8]: Why is `${nightly}` needed?

Comment (Diegiwg) [reply packages/hakuneko-deb/hakuneko-deb.pacscript:8]: To have only one update point when the program changes versions. The URL of this repository changes in addition to the value of ${version}, that part which is in ${nightly} every time there is a new release.

**Comment by Diegiwg** [comment]
Now that you've passed all the tests, what are the next steps to publish in the repo?

**Comment by Henryws** [comment]
We wait for the Hakuneko to respond. If they don't within a couple days, we can merge it as-is.

**Comment by Henryws** [comment]
Please don't merge from master unless you are actively planning on updating the pacscript along with it, because it wastes our CI's time for something we already know works.[^1]

[^1]: https://github.com/pacstall/pacstall-programs/blob/master/CONTRIBUTING.md#pr-etiquette

**Comment by Diegiwg** [comment]
Oops, sorry about that. Thank you for the warning.

**Comment by Henryws** [comment]
I merged in master for testing purposes on #3666.

**Comment by Diegiwg** [comment]
> We wait for the Hakuneko to respond. If they don't within a couple days, we can merge it as-is.

I believe it is released. Based on a quote from the main developer, which was exposed as a third-party response to my Issue: <https://github.com/manga-download/hakuneko/issues/5881#issuecomment-1584433707>

File: packages/hakuneko-deb/hakuneko-deb.pacscript [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+  i386)
+    url="https://github.com/manga-download/hakuneko/releases/download/nightly-${nightly}/hakuneko-desktop_${version}_linux_${CARCH}.deb"
+    hash="3608ff4e8a3257fffedfa2dcea6501cb6d0d203235a2e4c6e6bbf01b93dce240"
+    ;;
+  *) return 1 ;;
+esac
+
+homepage="https://github.com/manga-download/hakuneko"
+description="HakuNeko is a cross-platform downloader for manga and anime from various websites"
+maintainer="Diegiwg <diegiwg@gmail.com>"
```
Comment (Henryws) [inline packages/hakuneko-deb/hakuneko-deb.pacscript:33]: Usually these variables go near the top, above the `case` statement.

**Review by Henryws — APPROVED** [review]

**Pull request merged**